### PR TITLE
Bump dockerfile hash to fix SARIF semVer

### PR DESCRIPTION
**Pull request by laurentsimon**
See https://github.com/ossf/scorecard/issues/1479 for details

### CODE REVIEW

**Comment by laurentsimon** [comment]
FYI, need to update the hash with the new v4.0.1 scorecard release.

File: Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #           -e INPUT_REPO_TOKEN=$GITHUB_AUTH_TOKEN \
 #           -e GITHUB_REPOSITORY="ossf/scorecard" \
 #           laurentsimon/scorecard-action:latest
-FROM gcr.io/openssf/scorecard@sha256:7def80b7386d0addfa20b408fe23eacf85280806dc1981782f4e7766c2eb8833 as base
+FROM gcr.io/openssf/scorecard@sha256:a6db69ae36f3199ada709fe86dd9a226d75fa1d20397326006473f54d60f4165 as base
```
Comment (azeemshaikh38) [inline Dockerfile:24]: Just cut a `v4.0.1` release on Scorecard. Will wait to verify that the new hash is unchanged before approving.

Comment (laurentsimon) [reply Dockerfile:24]: I will change since we don't have reproducible builds yet. Also, the version embedded in scorecard binary will be bumped.

Comment (laurentsimon) [reply Dockerfile:24]: new hash is `8165ad910019422f40c51cbb97ff6e7db0e2e2e11faebf59e0b6f1a2eb66ebd7`

Comment (azeemshaikh38) [reply Dockerfile]: New image here: `gcr.io/openssf/scorecard@sha256:8165ad910019422f40c51cbb97ff6e7db0e2e2e11faebf59e0b6f1a2eb66ebd7`

Comment (laurentsimon) [reply Dockerfile]: updated in the PR. PTAL

**Review by azeemshaikh38 — APPROVED** [review]

**Pull request merged**